Composer/packages/client/src/components/Notifications/TunnelingSetupNotification.tsx

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx, css } from '@emotion/core';
+import React from 'react';
+import formatMessage from 'format-message';
+import { IconButton, IButtonStyles } from 'office-ui-fabric-react/lib/Button';
+import { NeutralColors, FontSizes } from '@uifabric/fluent-theme';
+import { Link } from 'office-ui-fabric-react/lib/Link';
+import { FontWeights } from '@uifabric/styling';
+
+import { platform, OS } from '../../utils/os';
+
+import { CardProps } from './NotificationCard';
+
+const container = css`
+  padding: 0 16px 16px 40px;
+  position: relative;
+`;
+
+const commandContainer = css`
+  display: flex;
+  flex-flow: row nowrap;
+  position: relative;
+  padding: 4px 28px 4px 8px;
+  background-color: ${NeutralColors.gray20};
+  line-height: 22px;
+  margin: 1rem 0;
+`;
+
+const copyContainer = css`
+  margin: 0;
+  margin-bottom: 4px;
+  font-size: ${FontSizes.size16};
+  font-weight: ${FontWeights.semibold};
+`;
+
+const copyIconColor = '#005A9E';
+const copyIconStyles: IButtonStyles = {
+  root: { position: 'absolute', right: 0, color: copyIconColor, height: '22px' },
+  rootHovered: { backgroundColor: 'transparent', color: copyIconColor },
+  rootPressed: { backgroundColor: 'transparent', color: copyIconColor },
+};
+
+const linkContainer = css`
+  margin: 0;
+`;
+
+const getNgrok = () => {
+  const os = platform();
+  if (os === OS.Windows) {
+    return 'ngrok.exe';
+  }
+
+  return 'ngrok';
+};
+
+export const TunnelingSetupNotification: React.FC<CardProps> = (props) => {
+  const { title, data } = props;
+  const port = data?.port;
+  const command = `${getNgrok()} http ${port} --host-header=localhost`;
+
+  const copyLocationToClipboard = async () => {
+    try {
+      await window.navigator.clipboard.writeText(command);
```

Comment:
does this need permission?

Reply:
Not for write: https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API